src/realm/object-store/c_api/config.cpp

```diff
@@ -22,6 +29,13 @@ RLM_API bool realm_config_set_schema(realm_config_t* config, const realm_schema_
     });
 }
 
+RLM_API uint64_t realm_config_get_schema_version(realm_config_t* config)
+{
+    return wrap_err([&]() {
+        return config->schema_version;
+    });
+}
```

Comment:
How is this method supposed to communicate to the caller that an error has occurred? Does a return value of 0 indicate the schema version is 0 or that an error has occurred?

Reply:
I don't believe an error can occur here, I've just wrapped it in `wrap_err` because we seem to do that across the board.

Reply:
Awesome, making it even less obvious to consumers when to check for errors and when not to 😅

Reply:
I brought this up on Simon's PR for his branch. The implicit rule is that if the function returns a bool, you have to check for errors.

I'd have preferred the explicit C style of an out error var, but he brought up the point that for languages that do have some a heavy FFI, it's an unreasonable performance hit.

Reply:
Yeah, I'm not convinced on that part either, but not willing to die on that hill. In any case, my suggestion would be not to use `wrap_err` for functions where the consumer is not expected to check for errors. I would rather get some obscure native exception/crash in the case the function actually throws than have the app continue working with incorrect data.

Reply:
We definitely only use `wrap_err()` for functions that can fail. There are pretty few of those, but this is one of them, so I would say remove the `wrap_err()`.

Let's not rehash arguments about the error handling strategy. There are pros and cons, and they are outlined in the design doc. :-)

Reply:
I will say though...

> The implicit rule is that if the function returns a bool, you have to check for errors.

It's also painstakingly documented for each and every function, so I'm not sure I would call it "implicit". ;-)

Reply:
Fair point 😆 I'll take it back.